framework/src/kernels/Kernel.C

```diff
@@ -94,8 +94,14 @@ Kernel::computeResidual()
 
   precalculateResidual();
   for (_i = 0; _i < _test.size(); _i++)
+  {
     for (_qp = 0; _qp < _qrule->n_points(); _qp++)
+    {
+      if (_local_ref.size())
```

Comment:
I'm not the biggest fan of adding an extra branch that is a nested for loop deep to every residual calc

Reply:
hey look at that, someone's looking out there in github land 😁 

I imagine there would be heartburn about this. Maybe there is another way? I will say this quick hack is not making previously unsolvable problems solvable, so I'm extremely interested in this.

Reply:
Hah I saw this a ways back - the things that involve tagging tend to stick out to me 😆 

My only "better" solution would be to throw the whole i, qp loop into a lambda and then have another branch that just does the whole loop again for the ref if needed. Not  _that_ much better.

Reply:
> My only "better" solution would be to throw the whole i, qp loop into a lambda and then have another branch that just does the whole loop again for the ref if needed. Not _that_ much better.

Would that avoid slowing down the current case where we're not computing this extra vector?

Reply:
What about:

```
if (_local_ref.size())
{
  for (_i = 0; _i < _test.size(); _i++)
  {
    for (_qp = 0; _qp < _qrule->n_points(); _qp++)
    {
      _local_ref(_i) += std::abs(_JxW[_qp] * _coord[_qp] * computeQpResidual());
      _local_re(_i) += _JxW[_qp] * _coord[_qp] * computeQpResidual();
    }
  }
}
else
  for (_i = 0; _i < _test.size(); _i++)
    for (_qp = 0; _qp < _qrule->n_points(); _qp++)
      _local_re(_i) += _JxW[_qp] * _coord[_qp] * computeQpResidual();
  ```

Reply:
No better suggestions...so I implemented this fix for now.

Reply:
What s the physical meaning of `std::abs(_JxW[_qp] * _coord[_qp] * computeQpResidual());` ?

The change of sign could be messing up a quadrature rule (with some negative weights) and arent the sum of the residuals not the sum of the absolute values of the residuals supposed to cancel at convergence?

Reply:
It is the absolute value of the magnitudes of the contributing reaction forces. The key is that this gives you a physics dependent reference value to converge with respect to.

Reply:
Turns out this isn't needed, and the absolute value residual contributions are accumulated in TaggingInterface.